Frameworks/CoreText/CTLine.mm

```diff
@@ -370,15 +374,22 @@ CGFloat CTLineGetOffsetForStringIndex(CTLineRef lineRef, CFIndex charIndex, CGFl
                                 run->_stringIndices.begin() - 1;
 
                     if (index >= 0) {
-                        ret = std::accumulate(run->_dwriteGlyphRun.glyphAdvances, run->_dwriteGlyphRun.glyphAdvances + index, ret);
+                        if (_GlyphRunIsRTL(run->_dwriteGlyphRun)) {
+                            ret += std::accumulate(run->_dwriteGlyphRun.glyphAdvances,
+                                                   run->_dwriteGlyphRun.glyphAdvances + index,
+                                                   run->_relativeXOffset,
+                                                   std::minus<CGFloat>());
+                        } else {
+                            ret += std::accumulate(run->_dwriteGlyphRun.glyphAdvances,
+                                                   run->_dwriteGlyphRun.glyphAdvances + index,
+                                                   run->_relativeXOffset);
+                        }
                     }
```

Comment:
nit: this is probably an awful idea, but you _could_ use a ternary for the last parameter instead of having the if/else